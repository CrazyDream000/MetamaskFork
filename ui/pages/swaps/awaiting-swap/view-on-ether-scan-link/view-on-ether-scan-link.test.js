import React from 'react';

import { renderWithProvider } from '../../../../../test/jest';
import ViewOnEtherScanLink from '.';

const createProps = (customProps = {}) => {
  return {
    txHash:
      '0x58e5a0fc7fbc849eddc100d44e86276168a8c7baaa5604e44ba6f5eb8ba1b7eb',
    blockExplorerUrl: 'https://block.explorer',
    isCustomBlockExplorerUrl: false,
    ...customProps,
  };
};

describe('ViewOnEtherScanLink', () => {
  it('renders the component with initial props', () => {
    const { container, getByText } = renderWithProvider(
      <ViewOnEtherScanLink {...createProps()} />,
    );
    expect(getByText('View on Etherscan')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders the component with a custom block explorer link', () => {
    const { container, getByText } = renderWithProvider(
      <ViewOnEtherScanLink
        {...createProps({
          blockExplorerUrl: 'https://custom-blockchain.explorer',
          isCustomBlockExplorerUrl: true,
        })}
      />,
    );
    expect(getByText('View at custom-blockchain.explorer')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
