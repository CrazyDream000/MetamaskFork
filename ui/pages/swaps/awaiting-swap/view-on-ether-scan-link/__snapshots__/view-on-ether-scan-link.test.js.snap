// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewOnEtherScanLink renders the component with a custom block explorer link 1`] = `
<div>
  <div
    class="awaiting-swap__view-on-etherscan awaiting-swap__view-on-etherscan--visible"
  >
    View at custom-blockchain.explorer
  </div>
</div>
`;

exports[`ViewOnEtherScanLink renders the component with initial props 1`] = `
<div>
  <div
    class="awaiting-swap__view-on-etherscan awaiting-swap__view-on-etherscan--visible"
  >
    View on Etherscan
  </div>
</div>
`;
